clcache.py

```diff
@@ -142,7 +144,7 @@ def __init__(self):
             os.makedirs(self.objectsDir)
         lockName = self.cacheDirectory().replace(':', '-').replace('\\', '-')
         timeout_ms = int(os.environ.get('CLCACHE_OBJECT_CACHE_TIMEOUT_MS', 10 * 1000))
-        self.lock = ObjectCacheLock(lockName, timeout_ms)
+        self.lock = SystemSharedLock(lockName, timeout_ms)
```

Comment:
Makes a lot of sense. By now, I think it would be senible to consider splitting the `clcache.py` file, there are a couple of generic utility classes by now. Both `SystemSharedLock{Exception}` as well as `PersistentJSONDict` seem to be good candidates.